src/utils/components/PodSelectorModal/selectorUtils.ts

```diff
@@ -0,0 +1,67 @@
+import { Selector } from '@openshift-console/dynamic-plugin-sdk';
+import { isEmpty } from '@utils/utils';
+
+import { selectorToString } from '../Selector/utilts';
+
+import { requirementFromString } from './selectorRequirements';
+import { FromRequirementsOptions, Requirement } from './types';
+
+export const arrayify = (obj: any): string[] =>
+  Object.entries(obj).map(([key, value]) => (value ? `${key}=${value}` : key));
+
+export const objectify = (arr: string[]): Record<string, any> =>
+  arr.reduce((acc, item) => {
+    const [key, value = null] = item.split('=');
+    acc[key] = value;
+
+    return acc;
+  }, {});
+
+export const split = (str: string) => (str.trim() ? str.split(/,(?![^(]*\))/) : []); // [''] -> []
+
+export const cleanSelectorStr = (tag: string) => selectorToString(selectorFromString(tag));
+export const cleanTags = (currentTags: string[]) => split(cleanSelectorStr(currentTags.join(',')));
+
+export const isTagValid = (tag: string, isBasic: boolean) => {
+  const requirement = requirementFromString(tag);
+  return !!(requirement && (!isBasic || requirement.operator === 'Equals'));
+};
+
+export const fromRequirements = (
+  requirements: Requirement[],
+  options: FromRequirementsOptions = {},
+): Selector => {
+  options = options;
+
+  if (options.undefinedWhenEmpty && isEmpty(requirements)) {
+    return;
+  }
+
+  const selector = requirements.reduce(
+    (acc, requirement) => {
+      if (requirement.operator === 'Equals') {
+        acc.matchLabels[requirement.key] = requirement.values[0];
```

Comment:
maybe add ?.

also instead of else u can simply return acc

Reply:
Can also use Operator.Equals instead of hard-coded 'Equals'